frontend/src/libs/components/search/search.tsx

```diff
@@ -0,0 +1,47 @@
+import { type SubmitHandler } from 'react-hook-form';
+
+import {
+  useAppForm,
+  useCallback,
+  useDebounce,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+
+import { DEFAULT_SEARCH_PAYLOAD } from './libs/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  onValueChange: (payload: { search: string }) => void;
+};
+
+const Search = ({ onValueChange }: Properties): JSX.Element => {
+  const { control, handleSubmit } = useAppForm({
+    defaultValues: DEFAULT_SEARCH_PAYLOAD,
+    mode: 'onChange',
+  });
+
+  const debouncedOnValueChange = useDebounce(onValueChange);
+
+  const handleFormChange = useCallback(
+    (event_: React.BaseSyntheticEvent): void => {
+      void handleSubmit(
+        debouncedOnValueChange as unknown as SubmitHandler<{ search: string }>,
+      )(event_);
+    },
+    [debouncedOnValueChange, handleSubmit],
+  );
+
+  const { field } = useFormController({ name: 'search', control });
+
+  return (
+    <form onChange={handleFormChange}>
+      <input
```

Comment:
Can we use our custom Input component here?

Reply:
@fshabanov  but there is default styles which are unsuitable for this page and i can`t use className as prop there

Reply:
@Ekit18 as I see the only difference in styles is the background color, right?

Reply:
@fshabanov not only but it is the beggest difference

Reply:
In that case let's leave it as it is for now